### PR TITLE
feat(trtexec_vendor): remove package

### PR DESCRIPTION
**Pull request by kminoda**
## Description
Remove this package since this is no longer used in ml_model_provider or any other packages in Autoware Universe.

Furthermore, this package includes git clone during CMake, which makes the current `build-and-test` CI fail. 

## Tests performed

Should be OK as long as the build succeeds in CI.

## Effects on system behavior

No effect expected.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/detected_object_validation/src/object_lanelet_filter.cpp [hunk 1]
```diff
@@ -165,6 +165,8 @@ lanelet::ConstLanelets ObjectLaneletFilterNode::getIntersectedLanelets(
   const LinearRing2d & convex_hull, const lanelet::ConstLanelets & road_lanelets)
 {
   lanelet::ConstLanelets intersected_lanelets;
+
```
Comment (wep21) [inline perception/detected_object_validation/src/object_lanelet_filter.cpp:168]: This is irrelevant comment.

**Review by wep21 — APPROVED** [review]
LGTM

**Comment by wep21** [comment]
@felixf4xu @xmfcx Our internal process doesn't use `trtexec` anymore. Is it possible to remove this package?

**Comment by felixf4xu** [comment]
> @felixf4xu @xmfcx Our internal process doesn't use `trtexec` anymore. Is it possible to remove this package?

I use autoware.auto as it has, so I'm fine to remove it.

Actually, even outside autoware, I always build and install trtexec with tensorrt installation, I agree it should be removed from autoware.

**Review by xmfcx — APPROVED** [review]
Let's remove it then, if it becomes necessary, we can just revert it later.

**Pull request merged**